tests/testthat/test-utility-extra.R

```diff
@@ -0,0 +1,400 @@
+
+# Current pedestrian
+p1 = matrix(c(0, 0), 1, 2)
+rownames(p1) = nms[1]
+
+# All pedestrians
+p2 = rbind(c(0, 0), c(0.5, 0.5), c(1, 1))
+rownames(p2) = nms
+
+# Pedestrian angle in front
+a_front = 90
+names(a_front) = nms[1]
+
+# Pedestrian angles not in front
+a_not_front = 135
+names(a_not_front) = nms[1]
+
+# Pedestrian angles vector
+a = c(90, 90, 90)
+names(a) = nms
+
+# Pedestrian velocities
+v = c(0.5, 0.5, 0.5)
+names(v) = nms
+
+# Goals
+P1 = matrix(c(2, 2, -1, -1), 2, 2)
+rownames(P1) = c('g1', 'g2')
+attr(P1, 'i') = 1
+
+# Pedestrian radius
+r = c(0.5, 0.5, 0.5)
+names(r) = nms
+
+# Pedestrian group
+group = c(2, 1, 2)
+names(group) = nms
+
+# Cell centres
+set.seed(23123)
+centres = matrix(rnorm(66), 33, 2)
+
+# Predicted pedestrian positions
+p_pred = rbind(c(0, 0), c(0.75, 0.75), c(1.25, 1.25))
+rownames(p_pred) = nms
+
+# Objects not occluding view between p1 and p2
+objects = list(
+  list(x = c(1, 0), y = c(0, 1)),
+  list(x = c(0, 1), y = c(1, 0))
+)
+
+# Objects occluding view between p1 and p2
+objects_occlude = list(
+  list(x = c(0.15, 0.35), y = c(0.15, 0.35))
+)
+
+# State list
+state = list(
+  p = p2,
+  a = c(45, 45, 45),
+  v = v,
+  r = r,
+  P = list(
+    P1, P1, P1
+  ),
+  group = group
+)
+names(state$a) = nms
+
+# Pos of current pedestrian in p2
+n = 1
+
+testthat::test_that("Destination angle computation works", {
+  ref = m4ma::destinationAngle_r(a_front, p1, P1)
+  res = m4ma::destinationAngle_rcpp(a_front, p1, P1)
+  testthat::expect_equal(res, ref)
+})
+
+testthat::test_that(
+  "Predicted distance works (when all pedestrians are seen in front)", {
+    ref = m4ma::predClose_r(n, p1, a_front,
+                            p2, r, centres, p_pred, objects)
+    res = m4ma::predClose_rcpp(n-1, p1, a_front,
+                               p2, r, centres, p_pred, objects)
+    testthat::expect_equal(res, ref)
+  }
+)
+
+testthat::test_that(
+  "Predicted distance works (when one or less pedestrians are occluded)", {
+    ref = m4ma::predClose_r(n, p1, a_front,
+                            p2, r, centres, p_pred, objects_occlude)
+    res = m4ma::predClose_rcpp(n-1, p1, a_front,
+                               p2, r, centres, p_pred, objects_occlude)
+    testthat::expect_equal(res, ref)
+    testthat::expect_equal(res, NULL)
+  }
+)
+
+testthat::test_that(
+  "Predicted distance works (when pedestrians are seen but not in front)", {
+    ref = m4ma::predClose_r(n, p1, a_not_front,
+                            p2, r, centres, p_pred, objects)
+    res = m4ma::predClose_rcpp(n-1, p1, a_not_front,
+                               p2, r, centres, p_pred, objects_occlude)
+    testthat::expect_equal(res, ref)
+    testthat::expect_equal(res, NULL)
+  }
+)
+
+testthat::test_that("Egocentric objects computation works", {
+  ref = m4ma::eObjects_r(p1, p2, r)
+  ref_list = list(ac = ref[, , 1], cw = ref[, , 2])
+  res = m4ma::eObjects_rcpp(p1, p2, r)
+  names(dimnames(res$ac)) = c('', '') # C++ doesn't name dims

```

Comment:
I don't understand why you set dimnames here. The comment says that C++ doesn't name dims, however Rcpp does give you this opportunity by using rownames() and colnames(). For example, see: https://stackoverflow.com/questions/28521175/setting-only-column-names-in-rcpp

Reply:
I could indeed have set "empty" (i.e., `''`) dimnames for the matrix here. However, this function is only called by one other function which does not require dimnames, so I thought the function would be faster if I didn't add them. If you think that it is confusing that the functions don't produce exactly the same result, I can add them!

Reply:
I don't think that is confusing. From the perspective of the goal for which we've been hired, that is: to match as close as possible the output of the functions we're translating from R to c++, this is not achieving that. Of course, sometimes we don't manage to 100% achieve this goal for constraints related to c++, but in this case, the constraint is just not there. You say that you did it because you thought it was going to be faster, but is it? and even if so, isn't it (of course) going to be faster because you are skipping a step?

Reply:
Yes, I am skipping a step which is apparently not needed. But it might backfire in the future, so I will add the dimnames!